plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java

```diff
@@ -788,6 +788,9 @@ public KubernetesClusterVO doInTransaction(TransactionStatus status) {
                 if (autoscaleEnabled != null) {
                     updatedCluster.setAutoscalingEnabled(autoscaleEnabled.booleanValue());
                 }
+                if (state != null) {
```

Comment:
@harikrishna-patnala curious, why do we need to update the state manually? I feel that should be updated by FSM.
Also, I don't see a real use where code is explicitly updating the state with this.

Reply:
yes, this is not required, kind of duplicate update code since we have fixing that in update query. Reverted this change.